### PR TITLE
Improve String pattern to support both " and ' 

### PR DESCRIPTION
**Pull request by m-anshuman2166**
Adds support for highlighting of both type of strings

### CODE REVIEW

**Comment by m-anshuman2166** [comment]
Also fixes the issue in which content between 2 strings gets highlighted issue :

image

**Review by AmrDeveloper — CHANGES_REQUESTED** [review]
Hello @m-anshuman2166, 

Your new regex looks great I have tested it and it works but we have one pattern for char which is as you called string but with a single quote and another pattern for a string with double quotes so people can take what they want and can also add a different color for each one, so can you split your pattern into 2 patterns one for PATTERN_CHAR `['](.*?)[']` and other for PATTERN_STRING `[\"](.*?)[\"]` in every language file

**Comment by m-anshuman2166** [comment]
Okay, let me edit

File: app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java [hunk 1]
```diff
@@ -34,8 +34,8 @@ public class JavaLanguage {
     private static final Pattern PATTERN_ANNOTATION = Pattern.compile("@.[a-zA-Z0-9]+");
     private static final Pattern PATTERN_TODO_COMMENT = Pattern.compile("//TODO[^\n]*");
     private static final Pattern PATTERN_NUMBERS = Pattern.compile("\\b(\\d*[.]?\\d+)\\b");
-    private static final Pattern PATTERN_CHAR = Pattern.compile("'[a-zA-Z]'");
-    private static final Pattern PATTERN_STRING = Pattern.compile("['](.*?)[']|[\"](.*?)[\"]");
+    private static final Pattern PATTERN_CHAR = Pattern.compile("['](.*?)[']|");
```
Comment (AmrDeveloper) [inline app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java:37]: Good, but you should remove the or `|` from the PATTERN_CHAR to be `['](.*?)[']`

Comment (AmrDeveloper) [reply app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java:37]: You can also update the 3 files in one commit no need for 3

**Review by AmrDeveloper — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java (hunk 1 above)
Comment (m-anshuman2166) [reply app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java:37]: See now....

Comment (AmrDeveloper) [reply app/src/main/java/com/amrdeveloper/codeviewlibrary/syntax/JavaLanguage.java:37]: You need to push new commit to resolve them, they are the same now

**Comment by m-anshuman2166** [comment]
Lol, seems like i needa sleep 

`Repeating one mistake 10 times`

**Review by AmrDeveloper — APPROVED** [review]

**Comment by AmrDeveloper** [comment]
Thanks for your PR, @m-anshuman2166 keep going bro

**Pull request merged**